File: algorithms/0540.single_element_in_a_sorted_array/single_element_in_a_sorted_array.rs
impl Solution {
    pub fn single_non_duplicate(nums: Vec<i32>) -> i32 {
        let (mut lo, mut hi) = (0, nums.len() - 1);
        while lo < hi {
            let mut mi = (hi - lo) / 2 + lo;
            if mi % 2 == 1 {
                mi -= 1;
            }
            if nums[mi] == nums[mi + 1] {
                lo = mi + 2;
            } else {
                hi = mi;
            }
        }
        nums[lo]
    }
}

File: algorithms/0540.single_element_in_a_sorted_array/single_element_in_a_sorted_array.cpp
class Solution {
public:
    int singleNonDuplicate(const vector<int>& nums) {
        int lo = 0, hi = nums.size() - 1;
        while (lo < hi) {
            const int mi = lo + (hi - lo) / 2;
            if (nums[mi] == nums[mi ^ 1]) {
                lo = mi + 1;
            } else {
                hi = mi;
            }
        }
        return nums[hi];
    }
};